File: cheat-sheets/javaScript/array.js
console.log("Array info")

arr = []
numbers = [1,2,3,4,5]
numbers2 = [1,2,3,4,5]

var add = function (num) {
    num + 1
    return
}

arr.forEach(add) //loop through the array


// array methods

arr.join('') // turn array into string using the separator
arr.concat(numbers2)// create new array joining two array into one
arr.indexOf("string") // return index of a certain element, if not available return -1
arr.pop(0) // remove element using index number
arr.push()// push what ever data into the last position
arr.shift(0)// remove element from start of the array
arr.unshift()// add element to start of the array
arr.slice(0,3)//create new array. include and not include.
arr.sort(fn)// converts to string - default ascending order
arr.reverse()// reverse the order of the array
arr.includes("element")//return boolean value if in array.
arr.filter( fn() )// new array, if a condition is evaluated true
Array.isArray(arr) // check to see if array is an array boolean

//=================
//  ARRAY LOOP
//=================
arr.forEach(function(elem, index, array) {
    console.log(`elem:${elem}, index: ${index}, arry: ${array}`)
    
    })

arr.forEach(elem => {
    console.log(elem)
});

File: cheat-sheets/Ruby/array.rb
numbers = [1,2,3,4,5]
words = ["hello", "what", "is", "up", "dude"]

numbers.push[6]
number << 6 # also push

numbers.sort # accending order
numbers.sort.reverse # decending order

.first # get the first element
.last # get the last element

numbers + words # concat 2 arrays together making one big array

.shuffle # randomize the order of the elements in the arary

.sample # pick a random element from the array
.to_a - # create an array from data
.uniq # remove duplicates from array

=====================
        EACH
=====================
#loop through arrays
.each #for each element do 
numbers.each do |elem| # pass in like a function
puts elem
end
=====================
        SELECT # needs a block of code. true or false
=====================
.select
numbers.select { |n| n.even? }

words.select { |elem, index| index.even? }

words.select { |elem| elem.start_with? "a"}